test/Constraints/sr7884.swift

```diff
@@ -0,0 +1,19 @@
+// RUN: %target-typecheck-verify-swift
+
+// SR-7884
+func f<T>(_ x: T) -> T {
+  return x
+}
+
+func f<T>(_ x: T?) -> T? {
+  return x
+}
+
+let r = f(1) // calls optional variant
```

Comment:
shouldn't this read "calls non-optional variant"?

Reply:
Yeah, that was a holdover from the original test case. I'll remove the comment.